src/coreclr/src/System.Private.CoreLib/src/System/TypedReference.cs

```diff
@@ -7,15 +7,16 @@
 
 using System.Reflection;
 using System.Runtime.CompilerServices;
+using Internal.Runtime.CompilerServices;
 
 namespace System
 {
     [CLSCompliant(false)]
     [System.Runtime.Versioning.NonVersionable] // This only applies to field layout
     public ref struct TypedReference
     {
-        private IntPtr Value;
-        private IntPtr Type;
+        private readonly ByReference<byte> _value;
+        private readonly IntPtr _type;
 
```

Comment:
Could this be updated to be `RuntimeTypeHandle` instead?

Reply:
What would be the advantage?

Note that `RuntimeTypeHandle` is a wrapper over a `System.Type` instance:

https://github.com/dotnet/runtime/blob/2cf4433f64cd10070f19de0c091295108a982b7f/src/coreclr/src/System.Private.CoreLib/src/System/RuntimeHandles.cs#L77-L78

This `IntPtr` points to the `MethodTable`.

Reply:
Argh, sorry, that’s right. What I was trying to say is that making that a `MethodTable*` field would make it easier in the future to make TypedReference actually useful, e.g. if we wanted to implement a way to get the type size.

Reply:
That shouldn't be a problem to do when we need that - IntPtr and MethodTable* are not very different for the JIT/VM - but right now the field is transparent to TypedReference and IntPtr conveys that. Leaving it at IntPtr makes it more likely we can share this file with Mono (although weirdly, Mono has three fields here so we can't unify just yet).